app/main.py

```diff
@@ -1,3 +1,14 @@
 def cache(func):
-    # Write your code here
-    pass
+    dictionary = {}
+
+    def wrapper(*args):
+        if args in dictionary.keys():
+            print('Getting from cache')
+            return dictionary[args]
+        else:
+            res = func(*args)
+            dictionary[args] = res
+            print('Calculating new result')
+            return res
```

Comment:
You can use a single `return` statement out of the `if/else` statement instead of two inside it.

Reply:
Also, `result` is much better than `res`